### PR TITLE
Investigate ingesting Open Street Maps

### LINKED ISSUE #13: Investigate ingesting Openstreetmap
- https://wiki.openstreetmap.org/wiki/OSM_Semantic_Network (seems quite old)
- https://github.com/GeoKnow/LinkedGeoData:
	- repo seems more recent (commits are 5 months old)
	- [website](https://linkedgeodata.org/)
	- their [releases](https://hobbitdata.informatik.uni-leipzig.de/LinkedGeoData/downloads.linkedgeodata.org/releases/) seem very old too, but hopefully we can run their code to get an up-to-date RDF
	- we could eventually experiment with their released graphs first

### PR DESCRIPTION
**Pull request by doogyb**
closes #13

### CODE REVIEW

**Comment by doogyb** [comment]
@mindrones I need to write a short script to automate the downloading of the zipped RDF files. Where in the dir structure should I put this script? My guess is:

```
ontologies
  OSM
    queries
      stress
        ...
      explore
        ...
    results
      stress
        ...
      explore
        ...
    bin
      fetch.js
...  
```

**Comment by mindrones** [comment]
I'd say:

```
/ontologies
	OSM/
		queries/
			stress/
			explore/
		results/
			stress/
			explore/
	.../
/src
	bin/
		OSM/
			fetch.js
		.../
	utils/
		...
```

**Comment by mindrones** [comment]
We could have a `data` dir within each ontology dir so that e.g. `src/bin/OSM/fetch.js` would put the fetched data in `ontologies/OSM/data/*`:

```
/ontologies
	OSM/
		data/
		queries/
			stress/
			explore/
		results/
			stress/
			explore/
	.../
/src
	bin/
		OSM/
			fetch.js
		.../
	utils/
		...
```

**Comment by doogyb** [comment]
I've left it for now, as the data won't be stored on the repo either way, and the script in question maintains the flexibility of being able to output the files to any given directory by supplying the path as an argument.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (mindrones) [inline package.json:1]: we usually sort package.json keys alphabetically

File: package.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "semantic",
+    "web",
+    "linked",
+    "data",
+    "rdf",
+    "virtuoso",
+    "neptune",
+    "owl",
+    "turtle",
+    "sparql"
```
Comment (mindrones) [inline package.json]: I'd sort these alphabetically

File: package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  ],
+  "author": "",
+  "license": "ISC",
```
Comment (mindrones) [inline package.json]: these two can go as we have a license in the repo

File: package.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "ontologies",
+  "version": "1.0.0",
+  "description": "A repository for tracking work done on ontologies at Nesta. Mainly for creating issues on GitHub and saving interesting queries, as well as documenting the work and progress being made on Ontologies and using Linked Data sources.",
+  "main": "index.js",
+  "type": "module",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (mindrones) [inline package.json:8]: Here we need to have an npm script that runs `src/bin/OSM/download_latest_lgd_releases.sh`

**Review by mindrones — CHANGES_REQUESTED** [review]
left some minor comments

File: package.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  "name": "ontologies",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/nestauk/ontologies.git"
+  },
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "type": "module",
+  "version": "1.0.0"
```
Comment (mindrones) [inline package.json]: for now I'd set the version to `0.0.1`

File: src/bin/linkedGeoData/fetch.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
+if [ $# -ne 1 ] || [ ! -d $1 ]; then
```
Comment (mindrones) [inline src/bin/linkedGeoData/fetch.sh]: I'd rename this to `fetchReleases.sh`

File: package.json [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  ],
+  "main": "index.js",
+  "name": "ontologies",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/nestauk/ontologies.git"
+  },
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
```
Comment (mindrones) [inline package.json]: I'd use `scripts` to run executables in `bin/`

```suggestion
  "scripts": {
	"lgdFetchReleases": "bash src/bin/linkedGeoData/fetchReleases.sh",
	"lgdFetchURLs": "node src/bin/linkedGeoData/fetchURLs.js"
  },
```

File: src/bin/linkedGeoData/fetch_urls.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import fetch from 'node-fetch';
```
Comment (mindrones) [inline src/bin/linkedGeoData/fetch_urls.js]: I'd rename to `src/bin/linkedGeoData/fetchURLs.js`

File: .eslintrc.yml [hunk 9]
```diff
@@ -0,0 +1,268 @@
+env:
```
Comment (mindrones) [inline .eslintrc.yml]: I'd move this to `src/`

File: package.json [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    "ontologies",
+    "owl",
+    "rdf",
+    "semantic",
+    "sparql",
+    "turtle",
+    "virtuoso",
+    "web"
+  ],
+  "main": "index.js",
```
Comment (mindrones) [inline package.json]: I'd remove `main` for now

File: package-lock.json [hunk 11]
```diff
@@ -0,0 +1,2129 @@
+{
```
Comment (mindrones) [inline package-lock.json:1]: I'd run `npm install` in the root to make sure this is consistent with the changes in `package.json`

**Review by mindrones — CHANGES_REQUESTED** [review]
first pass

**Review by mindrones — APPROVED** [review]
Great, good to merge 👍

**Comment by mindrones** [comment]
On second thought, I’d add `closes #17` to the commit message so we can get rid of #17 too!

**Comment by doogyb** [comment]
closes #17

**Comment by doogyb** [comment]
closes #17

**Pull request merged**